INPTPZ1/FractalSolver.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Numerics;
+using INPTPZ1.Mathematics;
+
+namespace INPTPZ1
+{
+	class FractalSolver
+	{
+		private readonly string outputPath;
+		private PointF minimum;
+		private PointF maximum;
+		private readonly ImageHandler imageHandler;
+		private readonly ResolutionHolder resolutionHolder;
+
+		public FractalSolver(ResolutionHolder resolutionHolder, PointF minimum, PointF maximum, string outputPath)
+		{
+			this.resolutionHolder = resolutionHolder;
+			imageHandler = new ImageHandler(resolutionHolder);
+			this.minimum = minimum;
+			this.maximum = maximum;
+			this.outputPath = outputPath;
+		}
+
+		internal void Solve()
+		{
+			List<Complex> koreny = new List<Complex>();
+
+			Polynom originalPolynom = InitOriginalPolynom();
+			Polynom derivedPolynom = originalPolynom.Derive();
+
+			double xstep = (maximum.X - minimum.X) / resolutionHolder.Width;
+			double ystep = (maximum.Y - minimum.Y) / resolutionHolder.Height;
+
+			for (int i = 0; i < resolutionHolder.Width; i++)
+			{
+				for (int j = 0; j < resolutionHolder.Height; j++)
+				{
+					double y = minimum.Y + i * ystep;
+					double x = minimum.X + j * xstep;
+
+					x = x == 0 ? 0.0001 : x;
+					y = y == 0 ? 0.0001f : y;
+
+					Complex actualComplex = new Complex(x, y);
+
+					float iteration = 0;
+					actualComplex = FindSolutionByNewtonIteration(originalPolynom, derivedPolynom, actualComplex, ref iteration);
+
+					imageHandler.SetPixel(j, i, ImageHandler.GetColorByParams(iteration, GetRootCount(koreny, actualComplex)));
+				}
+			}
+			imageHandler.SaveImage(outputPath);
+		}
+
+		private static Complex FindSolutionByNewtonIteration(Polynom originalPolynom, Polynom derivedPolynom, Complex actualComplex, ref float iteration)
```

Comment:
actualComplex is what? Its initial complex number, not actual. You're using one variable for two different purposes